### PR TITLE
Fix out-of-bounds read in ClockIdentity::set() caused by malformed an…

### PR DESCRIPTION
**Pull request by m46kruk**
Hello,

as discussed in [Issue 414](https://github.com/AVnu/Open-AVB/issues/414), here is my pull request.

Thank you and best regards
Matt

### CODE REVIEW

File: daemons/gptp/common/avbts_message.hpp [hunk 1]
```diff
@@ -360,8 +360,13 @@ class PathTraceTLV {
 	 * @param  buffer [in] Message buffer. It should be at least ::PTP_CLOCK_IDENTITY_LENGTH bytes long.
 	 * @return void
 	 */
-	void parseClockIdentity(uint8_t *buffer) {
-		int length = PLAT_ntohs(*((uint16_t *)buffer))/PTP_CLOCK_IDENTITY_LENGTH;
+	void parseClockIdentity(uint8_t *buffer, int size) {
```
Comment (andrew-elder) [inline daemons/gptp/common/avbts_message.hpp]: the added "size" parameter needs to be added to the doxygen comment block

**Comment by m46kruk** [comment]
Ah sorry, I've added the missing description to the doxygen comment block.
Also, I noticed that I wasn't keeping track of the buffer size correctly, so I've made sure to decrement the size whenever the pointer to the buffer is incremented.

**Comment by andrew-elder** [comment]
This looks ok to me. Anyone else?

**Comment by davidcemin** [comment]
looks good to me

**Pull request merged**